### DISCUSSION TITLE
Audio to text | Post-processing

### DISCUSSION THREAD

**Discussion by chenlung** [URUWorks/TeroSubtitler#16] [Ideas]
Tero Subtitler uses the models of Whisper for its audio-to-text feature, but the results are typically unsuitable for direct use.

As part of a post-processing feature in which the generated script is refined, what options would users like to see?

**Comment by GitGianluc** [comment]
Hi @chenlung, it’s been a long time since this discussion started, but – based on my experience using Tero – I can say that it would be a killer feature if we could synchronise the extracted subtitles with the speech segments, even if only partially.

As far as I understand, latest versions of Whisper.cpp are capable of performing this task using a component called “[Voice Activity Detection (VAD)](https://github.com/ggml-org/whisper.cpp#voice-activity-detection-vad)”: the VAD model used by Whisper.cpp is called [**Silero VAD**](https://github.com/snakers4/silero-vad).

I’d be keen to give it a go, but it’s a very technical matter that I can’t quite get to grips with; perhaps @URUWorks could help us out?
